src/dir_contents.ml

```diff
@@ -232,35 +232,56 @@ end = struct
         match (stanza : Stanza.t) with
         | Library lib ->
           let src_dir = d.ctx_dir in
-          let resolved = lazy (
-            let name = Library.best_name lib in
-            Lib.DB.find_even_when_hidden (Scope.libs scope) name
-            (* can't happen because this library is defined using the current
-               stanza *)
-            |> Option.value_exn
-          ) in
-          let kind : Modules_field_evaluator.kind =
-            match lib.implements, lib.virtual_modules with
-            | Some _, None ->
+          let kind, main_module_name, wrapped =
+            match lib.implements with
+            | None ->
+              let main_module_name =
+                match Library.main_module_name lib with
+                | This x -> x
+                | From _ -> assert false
```

Comment:
Could we add some comments for the assert falses?

I'm also forgetting why exactly they are present.

Reply:
Sure, I added a comment